x-pack/plugins/ml/public/application/components/color_range_legend/color_range_legend.tsx

```diff
@@ -12,7 +12,7 @@ import { EuiFlexGroup, EuiFlexItem, EuiText } from '@elastic/eui';
 const COLOR_RANGE_RESOLUTION = 10;
 
 interface ColorRangeLegendProps {
-  colorRange: (d: number) => string;
+  colorRange: (d: number) => number;
```

Comment:
This should stay `string`, the function takes in a number and returns a hex based color string like `#336699`. Did it trigger any errors when it was set to string?

Reply:
yes, it did
<img width="853" alt="Screenshot 2020-06-02 at 11 04 20" src="https://user-images.githubusercontent.com/3198181/83501749-dd32ee00-a4c0-11ea-82a6-c6f305a08f7c.png">
An inferred type of `colorRange` was:
```
const colorRange: (number[] & d3.scale.Linear<number, number>) | ((n: number) => number) | (number[] & d3.scale.Pow<number, number>)
```

I rolled back the changes. Could you fix the error, please?

Reply:
Thanks for clarifying, I was able to fix it in ffe3338. We were using the `d3` types incorrectly, was able to solve it as described here: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/8941#issuecomment-208517360